keras_core/layers/layer.py

```diff
@@ -602,6 +603,17 @@ def non_trainable_weights(self):
             return self.weights
         return [v for v in self.weights if not v.trainable]
 
+    @property
+    def metrics_variables(self):
```

Comment:
Does this work with compiled metrics? We should add support for that in the Trainer.

Reply:
The trainer override this method in https://github.com/keras-team/keras-core/blob/a465816da34c9ceebe3baa87f0e973b4bb751f86/keras_core/trainers/trainer.py#L233, which covered compiled metrics.

Reply:
Got it. Is it tested?

Reply:
Yes, the existing test in trainer does cover that.

Reply:
Eg in https://github.com/keras-team/keras-core/blob/a465816da34c9ceebe3baa87f0e973b4bb751f86/keras_core/trainers/trainer_test.py#L134